addons/mail/static/src/model/record_list.js

```diff
@@ -212,6 +212,10 @@ export class RecordListInternal {
     insert(recordList, val, fn, { inv = true, mode = "ADD" } = {}) {
         const inverse = getInverse(recordList);
         const targetModel = getTargetModel(recordList);
+        if (typeof val !== "object") {
+            // single-id data
+            val = { [recordList._store[targetModel].id]: val };
+        }
```

Comment:
Probably filter out `null` which has `typeof` also `"object"`.
Assigning `null` on a relational field is equivalent to assigning `false`/`undefined`, which is a clear:
https://github.com/odoo/odoo/blob/master/addons/mail/static/src/model/store_internal.js#L243

Reply:
I have added writing `null` at the end of each test and they all pass without any change in code.

Do you have any other case in mind?